lib/internal/streams/transform.js

```diff
@@ -244,3 +245,23 @@ Transform.prototype._read = function() {
     callback();
   }
 };
+
+Transform.by = function by(asyncGeneratorFn, opts) {
+  let _resolve;
+  let _promise = new Promise((resolve) => _resolve = resolve);
+  return from(Duplex, asyncGeneratorFn(async function*() {
+    while (true) {
+      const { chunk, done, cb } = await _promise;
+      if (done) return cb();
+      yield chunk;
+      _promise = new Promise((resolve) => _resolve = resolve);
+      cb();
```

Comment:
Can you do process.nextTick(cb) instead? errors thrown within this callback are getting wrapped in the async function otherwise.